types/jest/index.d.ts

```diff
@@ -342,7 +342,7 @@ declare namespace jest {
     }
 
     interface ExpectExtendMap {
-        [key: string]: (this: MatcherUtils, received: any, ...actual: any[]) => { message(): string, pass: boolean } | Promise<{ message(): string, pass: boolean }>;
```

Comment:
Functional change looks good! 

Two points that would help but I suppose aren't necessary for merging in:
* Please add back the tab, as not having consistent indentation makes it harder to read the definitions
* Unit tests?

_(I can do those in a subsequent PR if they're not done here)_

Reply:
`Tab` modification added. Sorry about that.

BTW, I think I should have done the same modification in the promise. Maybe for a future PR.